application-services/custom/influxdb-export/pkg/transforms/converions.go

```diff
@@ -0,0 +1,69 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2020 IOTech Systems LTD
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package transforms
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/edgexfoundry/app-functions-sdk-go/appcontext"
+	"github.com/edgexfoundry/go-mod-core-contracts/models"
+)
+
+type Conversion struct {
+}
+
+func NewConversion() Conversion {
+	return Conversion{}
+}
+
+func (f Conversion) TransformToInflux(edgexcontext *appcontext.Context, params ...interface{}) (continuePipeline bool, stringType interface{}) {
+	if len(params) < 1 {
+		return false, errors.New("No Event Received")
+	}
+
+	edgexcontext.LoggingClient.Debug("Transforming to InfluxDB Line Protocol format")
+
+	if event, ok := params[0].(models.Event); ok {
+
+		var buffer strings.Builder
+
+		// write device name as measurement
+		buffer.WriteString(event.Device)
+		// write tags if any, comma separated [,<tag_key>=<tag_value>[,<tag_key>=<tag_value>]]
+		for i, tag := range event.Tags {
```

Comment:
This doesn't produce what is in the comment. It will produce the following:

```
,tag<index>=<tag_key> <tag_value>, ,tag<index>=<tag_key> <tag_value>
```
suggest changing to:
```
          for _, tag := range event.Tags {
			buffer.WriteString(",")
			buffer.WriteString(tag)  // tag is the tag_key
			buffer.WriteString("=")
			buffer.WriteString(event.Tags[tag]) // this is the tag_value
		}
```

Reply:
Also a unit test would be helpful to validate the output of this function. yea, it is an example, but... ;-)

Reply:
I am just using the same nomenclature as Influx uses here:  https://docs.influxdata.com/influxdb/v2.0/reference/syntax/line-protocol/ in the comment.  It actually requires "tag"X as the key.  So keeping this as is in order to work.

As for testing - yes, I know but NGH today.

Reply:
Ok, so the comment then needs to change to reflect what the code is producing. Should be:
`,tag<index>=<tag_key> <tag_value>, tag<index>=<tag_key> <tag_value>`

Reply:
No - the tag key is the word "tag" plus the index.  So "tag1" which is the word tag plus 1 is the first tag key.
From the documentation below per InfluxDB
// Syntax
<measurement>[,<tag_key>=<tag_value>[,<tag_key>=<tag_value>]] <field_key>=<field_value>[,<field_key>=<field_value>] [<timestamp>]

// Example
myMeasurement,tag1=value1,tag2=value2 fieldKey="fieldValue" 1556813561098000000

Reply:
fixed per @lenny-intel suggestion